noir/noir-repo/noir_stdlib/src/field/mod.nr

```diff
@@ -211,6 +211,10 @@ pub comptime fn modulus_be_bytes() -> [u8] {}
 #[builtin(modulus_le_bytes)]
 pub comptime fn modulus_le_bytes() -> [u8] {}
 
+/// An unconstrained only built in to efficiently compare fields.
+#[builtin(field_less_than)]
+pub fn unconstrained_field_less_than(_x: Field, _y: Field) -> bool {}
```

Comment:
```suggestion
pub unconstrained fn field_less_than(_x: Field, _y: Field) -> bool {}
```

Reply:
Ah, didn't know I could mark builtins as unconstrained, will change

Reply:
I don't see why there would be a problem but I haven't checked myself.

Reply:
Seems to work perfectly

Reply:
Ahhh no, it seems like marking it with unconstrained makes you wrap it in an unsafe block but no unconstrained wrapper is created:
```noir
pub fn gt(a: Field, b: Field) -> bool {
    if is_unconstrained() {
        unsafe {
            field_less_than(b, a)
        }
    } else if a == b {
        false
    } else {
        // Take a hint of the comparison and verify it
        unsafe {
            if field_less_than(a, b) { // This one
                assert_gt(b, a);
                false
            } else {
                assert_gt(a, b);
                true
            }
        }
    }
}

```

```
acir(inline) fn gt f1 {
  b0(v0: Field, v1: Field):
    v6 = call is_unconstrained()
    jmpif v6 then: b8, else: b1
  b8():
    v14 = call field_less_than(v1, v0)
    jmp b9(v14)
  b9(v4: u1):
    return v4
  b1():
    v7 = eq v0, v1
    jmpif v7 then: b6, else: b2
  b6():
    jmp b7(u1 0)
  b7(v3: u1):
    jmp b9(v3)
  b2():
    v9 = call field_less_than(v0, v1)
    jmpif v9 then: b4, else: b3
  b4():
    call f2(v1, v0)
    jmp b5(u1 0)
  b5(v2: u1):
    jmp b7(v2)
  b3():
    call f2(v0, v1)
    jmp b5(u1 1)
}
```

What should I do? maybe prefix the builtin with __ to mark it as internal, remove the pub keyword and expose as pub an unconstrained wrapper of it?

Reply:
Yeah I think that exposing an unconstrained wrapper would be the best move.

Reply:
Actually, I don't think we need to make it pub at all. We already have a field lt exposed.

Reply:
field::bn254 uses it too so maybe `pub(crate)`

Reply:
Done